app/src/main/java/com/example/fina/data/repository/source/CoinDataSource.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.fina.data.repository.source
+
+import com.example.fina.data.model.Coin
+import com.example.fina.data.model.OrderBy
+import com.example.fina.data.model.OrderDirection
+import com.example.fina.data.model.PriceHistory
+import com.example.fina.data.model.TimePeriod
+import com.example.fina.data.repository.OnResultListener
+import com.example.fina.utils.Constant
+
+interface CoinDataSource {
+    interface Local {
+        fun getLocalCoins(listener: OnResultListener<MutableList<Coin>>)
```

Comment:
Dùng List thay vì MutableList em ơi, Data từ tầng Source trả về là cố định mà
```suggestion
        fun getLocalCoins(listener: OnResultListener<List<Coin>>)
```